### PR TITLE
Add check for cidr_block for existing vpc

### PR DESCRIPTION
**Pull request by shekhar-j**
# Description
Add check for cidr_block for existing vpc.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [X] I have maintained at least 80% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [X] Acceptance tests

### CODE REVIEW

File: apex/provider/resource_navigator_block_storage.go [hunk 1]
```diff
@@ -281,6 +290,21 @@ func (r *blockStorageResource) Create(ctx context.Context, req resource.CreateRe
 	}
 }
 
+// validateCreateParams validates the parameters for the create function.
+// It checks for any errors in the plan.
+// If there are errors, it returns the error.
+func (r *blockStorageResource) validateCreateParams(plan models.StorageModel) error {
```
Comment (doriac11) [inline apex/provider/resource_navigator_block_storage.go:296]: Should this be in the helper?

**Comment by shekhar-j** [comment]
run e2e test

**Review by gibarrama — APPROVED** [review]

**Review by nikitajoshi1 — APPROVED** [review]

**Review by doriac11 — APPROVED** [review]
I approve thanks for the updates! Please check the e2e tests.

**Comment by shekhar-j** [comment]
run e2e test

**Comment by shekhar-j** [comment]
run e2e test

**Comment by shekhar-j** [comment]
run e2e test

**Pull request merged**